Adding documentation about Endpoints write access in wake of CVE-2021-25740 (#29025)
kubernetes/website: content/en/docs/reference/access-authn-authz/rbac.md

  verbs: ["get", "list", "watch"]
```

#### Write Access for Endpoints

> 

Reviewer: I would place this content near the end of the doc, well after the "User-facing roles" section where the edit and admin roles are described (maybe just before the "Upgrading from ABAC" section?)

also, in the description of the `admin` and `edit` roles in the "User-facing roles" section, update the description to indicate write permissions to endpoints are not included in clusters created on 1.22+ and link to this section for details.

Author: Updated, thanks!

Kubernetes clusters created before Kubernetes v1.22 include write access to
Endpoints in the aggregated "edit" and "admin" roles. As a mitigation for
[CVE-2021-25740](https://github.com/kubernetes/kubernetes/issues/103675), this
access is not part of the aggregated roles in clusters that you create using
Kubernetes v1.22 or later.

Existing clusters that have been upgraded to Kubernetes v1.22 will not be
subject to this change. The [CVE
announcement](https://github.com/kubernetes/kubernetes/issues/103675) includes
guidance for restricting this access in existing clusters.

If you want new clusters to retain this level of access in the aggregated roles,
you can create the following ClusterRole:

> ```none
> {{</* codenew file="access/endpoints-aggregated.yaml" */>}}
> ```

Reviewer 2: This doesn't render how you expect.
Try:
```suggestion
{{< codenew file="access/endpoints-aggregated.yaml" >}}
```

Author: Updated, thanks!

  verbs: ["get", "list", "watch"]
```

> #### Write Access for Endpoints

Reviewer 2: nit:
```suggestion
#### Write access for Endpoints
```

Author: Updated, thanks!

Reviewer 2: I think that's still using title case (we prefer sentence case). Only a nit anyway, mind.

Author: 🤦 should be updated now, thanks!

Kubernetes clusters created before Kubernetes 1.22 include write access to
Endpoints in the aggregated "edit" and "admin" roles. As a mitigation for
[CVE-2021-25740](https://github.com/kubernetes/kubernetes/issues/103675), this

> access will not be part of the aggregated roles in newly created clusters using
> Kubernetes 1.22+.

Reviewer 2: ```suggestion
access is part of the aggregated roles in clusters that you create using Kubernetes
v1.22 or later.
```

Reviewer: explicitly say that a cluster upgraded from 1.21 or earlier will still include this permission in the aggregated edit and admin roles unless manually removed using the instructions in https://github.com/kubernetes/kubernetes/issues/103675

Reviewer 2: Even better: add a new task page explaining how to revise those aggregated ClusterRoles _and_ propose the fix for backport.
(feel free to raise an issue rather than doing it in this PR).

Author: Updated, thanks! Created https://github.com/kubernetes/website/issues/29044 to track a task page.

If you want new clusters to retain this level of access in the aggregated roles,
you can create the following ClusterRole:

```yaml

> apiVersion: rbac.authorization.k8s.io/v1

Reviewer 2: I would make this be a downloadable example that uses the `codenew` shortcode.

Author: Updated, thanks!

kind: ClusterRole
metadata:
  annotations:
    kubernetes.io/description: |-
      Add endpoints write permissions to the edit and admin roles. This was
      removed by default in 1.22 because of CVE-2021-25740. See

>       http://issue.k8s.io/103675. This can allow endpoint writers to direct

Reviewer 2: ```suggestion
      https://issue.k8s.io/103675. This can allow endpoints writers to direct
```
for encryption in transit and because `kind: Endpoints` is a plural noun

Author: Updated, thanks!

>       LoadBalancer or Ingress implementations to expose backend IPs that would

Reviewer 2: ```suggestion
       load balancers or similar components to expose backend IPs that would
```

Author: I think it's important to mention Ingress implementations here as not everyone may draw the connection between LB and Ingress.

      not otherwise be accessible, and can circumvent network policies or
      security controls intended to prevent/isolate access to those backends.
  labels:
    rbac.authorization.k8s.io/aggregate-to-edit: "true"

>   name: custom:aggregate-to-edit:endpoints

Reviewer 2: ```suggestion
  name: custom:aggregate-to-edit:endpoints # you can change this if you wish
```

Author: Updated, thanks!

rules:
  - apiGroups: [""]
    resources: ["endpoints"]

>     verbs: ["create", "delete", "deletecollection", "patch", "update"]

Reviewer: would you not also add in endpointslice permissions? if they need endpoints permissions, don't they also likely need endpointslice permissions?

Author: I've tried to keep this focused on restoring the status quo, not adding new access. EndpointSlices were never included in aggregated roles in a released Kubernetes version.

      --group=system:serviceaccounts
    ```

## Write access for Endpoints

Kubernetes clusters created before Kubernetes v1.22 include write access to
Endpoints in the aggregated "edit" and "admin" roles. As a mitigation for
[CVE-2021-25740](https://github.com/kubernetes/kubernetes/issues/103675), this
access is not part of the aggregated roles in clusters that you create using
Kubernetes v1.22 or later.

Existing clusters that have been upgraded to Kubernetes v1.22 will not be
subject to this change. The [CVE
announcement](https://github.com/kubernetes/kubernetes/issues/103675) includes
guidance for restricting this access in existing clusters.

If you want new clusters to retain this level of access in the aggregated roles,
you can create the following ClusterRole:

> ```none
> {{< codenew file="access/endpoints-aggregated.yaml" >}}
> ```

Reviewer 2: ```suggestion
{{< codenew file="access/endpoints-aggregated.yaml" >}}
```

Author: Woops, updated now, thanks!
